ksonnet-gen/ksonnet/emit.go

```diff
@@ -46,26 +87,59 @@ type root struct {
 
 	ksonnetLibSHA *string
 	k8sSHA        *string
+
+	isLegacySchema bool
 }
 
-func newRoot(spec *kubespec.APISpec, ksonnetLibSHA, k8sSHA *string) *root {
-	root := root{
+func newRoot(spec *kubespec.APISpec, ksonnetLibSHA, k8sSHA *string) (*root, error) {
+	uls, err := usesLegacySchema(spec.Info.Version)
+	if err != nil {
+		return nil, fmt.Errorf("check schema: %v", err)
+	}
+
+	r := root{
 		spec:         spec,
 		groups:       make(groupSet),
 		hiddenGroups: make(groupSet),
 
 		ksonnetLibSHA: ksonnetLibSHA,
 		k8sSHA:        k8sSHA,
+
+		isLegacySchema: uls,
+	}
+
+	crds := map[kubespec.DefinitionName]bool{}
+	for defName, def := range spec.Definitions {
+		if def.IsCRD() {
+			crds[defName] = false
+			ref := def.Properties["Schema"]
```

Comment:
I wrote this with a hardcoded string to confirm it works, but if I was writing it for prod, I'd handle this logic in the `SchemaDefinition` rather than relying on the emission routines to parse the schema definitions. Maybe worth replacing `SchemaDefinition#IsCRD` with something like `SchemaDefinition#CRDSchemaName() (DefinitionName, bool)` (or something) which would allow us to write the following (which I estimate to be clearer):

```
if crdDefName, isCRD := def.CRDSchemaName(); isCRD {
  crds[defName] = false
  crds[crdDefName] = false
}
```

So, this function would retrieve the `"Schema"` member, perform the `strings.TrimPrefix` below, and return as a `DefinitionName, true` (or `"", false` if not found).